lib/pbench/server/cache_manager.py

```diff
@@ -1442,6 +1442,7 @@ def create(self, tarfile_path: Path) -> Tarball:
             raise BadFilename(tarfile_path)
         name = Dataset.stem(tarfile_path)
         controller_name = None
+        metadata = None
         try:
             metadata = Tarball._get_metadata(tarfile_path)
             controller_name = metadata["run"]["controller"]
```

Comment:
This code (unfortunately, I can't include `except` clause body in my context) doesn't behave well in the error cases.  It should probably be split into two separate "operations" with separate `try` blocks:  if, for whatever reason, `_get_metadata()` returns `None`, then the resulting `TypeError` which will be reported at line 1451 ("'NoneType' object is not subscriptable") is pretty opaque.  It would be better if we could separate the "no controller" problem from the "no metadata" problem.

Reply:
Again, unless we're logging each, it makes no difference. I thought about carefully logging each alternative; but there's nothing we can do about it at ops review and I decided we really wouldn't benefit from the extra noise. Are you trying to argue otherwise?

Reply:
I'm suggesting that a future developer seeing "'NoneType' object is not subscriptable" might not think of it as an appropriate reason for "metadata.log is missing run.controller"...which is why I'm proposing skipping the return of `None`...which should allow for the proper error to be captured and displayed while at the same time getting rid of the `NoneType` problem.

Reply:
There's always a tension between under-logging and over-logging. (We spend more time filtering out uninteresting noise than tracking problems; and while I probably could have made the message more general I'm not sure that more detailed information really helps once we've identified that there's a problem with the dataset contents or structure.)

In specific, though, we were handling `CacheExtractBadPath`, but in one case what we got was `TarballUnpackError`. It wasn't caught in `_get_metadata`: so we "warned" here and continued, taking a server internal error at `tarball.metadata = metadata` because there *was* no `metadata` on that path.

I added the pre-assignment to avoid that; but it also pointed out that the `CacheExtractBadPath` was overly specific, and I just changed it to `Exception` since we weren't trying to tease out details at that level anyway. `_get_metadata` no longer propagates any exception, therefore the only possible exception in this try block is dereferencing the `dict` and there was no reason to complicate the code by breaking up the two statements.

However, yeah; I could have generalized the warning message, and I just didn't think of it at the time. I could also have explicitly checked for `not metadata`; but I think Pete would have argued that's "not Pythonic" (or at least I ended up internalizing his viewpoint that way 😆 ) and handling it as an exception avoided changing the existing code.

So, to get out of this:
1. I could just generalize the warning message;
2. I could remove the warning, just silently `pass` on an exception, and let the `has no controller name` become the generic message. (I very nearly did this, but I generally dislike an `except` with just a `pass`.);
3. I could break everything down step by step and issue a ton of specific warning messages that'll drive us crazy later;
4. or, as I did internally and implicitly while just hacking around on the weekend, I could say "problem solved, and the messaging isn't a big deal here, so pretend we're done". (Hey, I even wrote a unit test.)

OK, obviously you don't like (4), and that's fine: but are you arguing for detailed messaging in `_get_metadata` to explain `None`, or just generalizing the existing message here?

Reply:
>  I could also have explicitly checked for `not metadata`; but I think Pete would have argued that's "not Pythonic" (or at least I ended up internalizing his viewpoint that way 😆 )

Yeah, I was thinking similar things.  But, sometimes I think that the `try` construct is more cumbersome than doing the simple and obvious test.... 🦐